runtime/permissions.rs

```diff
@@ -63,515 +65,506 @@ impl Default for PermissionState {
+      .query(&Some(&(hostname, url.port_or_known_default())))
+      .check(&format!("network for \"{}\"", display_host), self.name)
   }
+}
 
-  pub fn request_hrtime(&mut self) -> PermissionState {
-    if self.hrtime == PermissionState::Prompt {
-      if permission_prompt("Deno requests access to high precision time") {
-        self.hrtime = PermissionState::Granted;
-      } else {
-        self.hrtime = PermissionState::Denied;
-      }
-    }
-    self.hrtime
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct BooleanPermission {
+  pub name: &'static str,
+  pub description: &'static str,
+  pub state: PermissionState,
+}
+
+impl BooleanPermission {
+  pub fn query(&self) -> PermissionState {
+    self.state
   }
 
-  pub fn revoke_read(&mut self, path: &Option<&Path>) -> PermissionState {
-    if let Some(path) = path {
-      let path = resolve_from_cwd(path).unwrap();
-      self
-        .read
-        .granted_list
-        .retain(|path_| !path_.starts_with(&path));
-    } else {
-      self.read.granted_list.clear();
-      if self.read.global_state == PermissionState::Granted {
-        self.read.global_state = PermissionState::Prompt;
+  pub fn request(&mut self) -> PermissionState {
+    if self.state == PermissionState::Prompt {
+      if permission_prompt(&format!("access to {}", self.description)) {
+        self.state = PermissionState::Granted;
+      } else {
+        self.state = PermissionState::Denied;
       }
     }
-    self.query_read(path)
+    self.state
   }
 
-  pub fn revoke_write(&mut self, path: &Option<&Path>) -> PermissionState {
-    if let Some(path) = path {
-      let path = resolve_from_cwd(path).unwrap();
-      self
-        .write
-        .granted_list
-        .retain(|path_| !path_.starts_with(&path));
-    } else {
-      self.write.granted_list.clear();
-      if self.write.global_state == PermissionState::Granted {
-        self.write.global_state = PermissionState::Prompt;
-      }
+  pub fn revoke(&mut self) -> PermissionState {
+    if self.state == PermissionState::Granted {
+      self.state = PermissionState::Prompt;
     }
-    self.query_write(path)
+    self.state
   }
 
-  pub fn revoke_net<T: AsRef<str>>(
-    &mut self,
-    host: &Option<&(T, Option<u16>)>,
-  ) -> PermissionState {
-    if let Some(host) = host {
-      self.net.granted_list.remove(&format_host(host));
-      if host.1.is_none() {
-        self
-          .net
-          .granted_list
-          .retain(|h| !h.starts_with(&format!("{}:", host.0.as_ref())));
-      }
-    } else {
-      self.net.granted_list.clear();
-      if self.net.global_state == PermissionState::Granted {
-        self.net.global_state = PermissionState::Prompt;
-      }
-    }
-    self.query_net(host)
+  pub fn check(&self) -> Result<(), AnyError> {
+    self.state.check(self.description, self.name)
   }
+}
+
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct Permissions {
+  pub read: UnaryPermission<ReadPermission>,
```

Comment:
I'm not sure why it's called `UnaryPermission` instead of `ListPermission`...

Reply:
> I'm not sure why it's called `UnaryPermission` instead of `ListPermission`...

Queries for that permission take one parameter.